src/angular/components/send/add-edit.component.ts

```diff
@@ -355,9 +360,16 @@ export class AddEditComponent implements OnInit {
     }
 
     typeChanged() {
-        if (!this.canAccessPremium && this.send.type === SendType.File && !this.premiumRequiredAlertShown) {
-            this.premiumRequiredAlertShown = true;
-            this.messagingService.send('premiumRequired');
+        if (this.send.type === SendType.File)
+        {
+            if (!this.canAccessPremium && !this.premiumRequiredAlertShown) {
+                this.premiumRequiredAlertShown = true;
+                this.messagingService.send('premiumRequired');
+            } else if (!this.emailVerified && !this.emailVerificationRequiredAlertShown &&
+                !this.premiumRequiredAlertShown) {
```

Comment:
I assume the `!this.premiumRequiredAlertShow` is to prevent multiple dialogs from appearing every time the user toggles it? If so I would suggest we switch to having a single property `alertShown` which both checks can use?

Reply:
Good point!